core/src/main/java/io/kestra/core/services/FlowService.java

```diff
@@ -39,6 +43,45 @@ public Stream<Flow> keepLastVersion(Stream<Flow> stream) {
         return keepLastVersionCollector(stream);
     }
 
+    public List<String> deprecationPaths(Flow flow) {
+        List<String> deprecationPaths = deprecationTraversal("", flow).toList();
+        return deprecationPaths.isEmpty() ? null : deprecationPaths;
+    }
+
+    private Stream<String> deprecationTraversal(String prefix, Object object) {
+        if(ClassUtils.isPrimitiveOrWrapper(object.getClass()) || String.class.equals(object.getClass())) {
+            return Stream.empty();
+        }
+
+        return Stream.concat(
+            object.getClass().isAnnotationPresent(Deprecated.class) ? Stream.of(prefix) : Stream.empty(),
+            Arrays.stream(object.getClass().getDeclaredFields())
+                .filter(f -> {
+                    int modifiers = f.getModifiers();
+                    return !Modifier.isStatic(modifiers);
+                })
+                .flatMap(field -> {
+                    try {
+                        field.setAccessible(true);
+                        Object fieldValue = field.get(object);
```

Comment:
Calling the getter is not a good idea, you should check the type of the field or the type of the getter.